packages/syft/src/syft/service/action/action_service.py

```diff
@@ -55,22 +135,73 @@ def np_array(self, context: AuthedServiceContext, data: Any) -> Any:
         np_pointer = self.set(context, np_obj)
         return np_pointer
 
+    def preprocess_action_to_store(
+        self,
+        context: AuthedServiceContext,
+        action_object: Union[ActionObject, TwinObject],
+    ):
+        if isinstance(action_object, ActionObject) and is_collection(
```

Comment:
the idea is here that when we do for instance client.lib.pandas.concat([df1,df2]), that this list that is sent before we execute the action does not contain a nested structure of ActionObject, but one ActionObject that contains the underlying [df1.syft_action_data, df2.syft_action_data ]

Reply:
This collection (list/dict/tuple/set) structure could be nested though, so we are searching through it. While we are searching we are also computing a few other things, like do we have read permission on the result